### PR TITLE
Add multiple axes unsqueeze and broadcasting to the right

### PR DESCRIPTION
**Pull request by fzimmermann89**
Both are not implemented in pytorch
https://github.com/pytorch/pytorch/issues/9410
https://github.com/pytorch/pytorch/issues/130529
https://github.com/pytorch/pytorch/issues/62148

I moved the usage in the signal models already over.
There might be more uses of this pattern in our code base and future uses. This implementation works with torch.jit.

### CODE REVIEW

**Comment by fzimmermann89** [comment]
This should be merged before #461

File: src/mrpro/utils/reshape.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        number of times to unsqueeze
+    """
+    return x.reshape(*x.shape, *(n * (1,)))
+
+
+def unsqueeze_left(x: torch.Tensor, n: int) -> torch.Tensor:
+    """Unsqueze multiple times in the leftmost dimension.
+
+    Example:
+        tensor with shape (1,2,3) and n=2 would result in tensor with shape (1,1,1,1,2,3)
```
Comment (ckolbPTB) [inline src/mrpro/utils/reshape.py]: I think there is one "1" too many in the resulting shape

File: tests/utils/test_reshape.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import torch
```
Comment (ckolbPTB) [inline tests/utils/test_reshape.py:3]: Tests should also have doc string as the first line

File: src/mrpro/utils/reshape.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
+"""Tensor reshaping utilities."""
```
Comment (ckolbPTB) [inline src/mrpro/utils/reshape.py:1]:
```
Returns
--------
```

is missing

**Review by ckolbPTB — APPROVED** [review]
just a few small things

File: src/mrpro/utils/reshape.py (hunk 1 above)
Comment (fzimmermann89) [inline src/mrpro/utils/reshape.py]:
```suggestion
        tensor with shape (1,2,3) and n=2 would result in tensor with shape (1,1,1,2,3)
```

**Comment by fzimmermann89** [comment]
pre-commit.ci autofix

**Pull request merged**